### ISSUE TITLE
《CAUSAL DISCOVERY WITH REINFORCEMENT LEARNING》

### ISSUE THREAD

**Issue by cczhangy** [huawei-noah/trustworthyAI#61]
When experimenting with the real dataset Sachs, the real dataset has been downloaded from the website, should I get the DAG.npy and data.npy of the real dataset?

**Comment by shaido987** [comment]
Yes, the data.npy and DAG.npy should be put into the same folder. Give the path as input when running main.py using `--data_path`. The relevant code to read the data is in [main.py and line 69-72](https://github.com/huawei-noah/trustworthyAI/blob/934a924a7849e9f6abf0f561f59a1211dca435ad/Causal_Structure_Learning/Causal_Discovery_RL/src/main.py#L69):

```python
if config.read_data:
    file_path = '{}/data.npy'.format(config.data_path)
    solution_path = '{}/DAG.npy'.format(config.data_path)
    training_set = DataGenerator_read_data(file_path, solution_path, config.normalize, config.transpose)
```

Note that the ground-truth in DAG.npy is only used for evaluation.

**Comment by shaido987** [comment]
The DAG.npy is simply a matrix with the true relations (available on the website as text or can be inferred from the graph there). For Sachs, this will be a 11x11 numpy matrix with 0s and 1s. Save this with `numpy.save` to get a npy file. 

For the data you can load the text file with `numpy.loadtxt`, skipping the first row, and then use `numpy.save` again to get the npy file.

**Comment by cczhangy** [comment]
OK, I see, thank you very much!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "huawei-noah/trustworthyAI"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月12日(星期二) 上午10:16
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [huawei-noah/trustworthyAI] 《CAUSAL DISCOVERY WITH REINFORCEMENT LEARNING》 (Issue #61)





 
The DAG.npy is simply a matrix with the true relations (available on the website as text or can be inferred from the graph there). For Sachs, this will be a 11x11 numpy matrix with 0s and 1s. Save this with numpy.save to get a npy file.
 
For the data you can load the text file with numpy.loadtxt, skipping the first row, and then use numpy.save again to get the npy file.

**Comment by shaido987** [comment]
No problems, happy to help.
